### PR TITLE
[#390] Increase Microsoft.Bot.Builder.AI.Orchestrator code coverage

### PR DESCRIPTION
**Pull request by MartinLuccanera**
Fixes # XXXX

## Description
This PR increases the code coverage for the `Microsoft.Bot.Builder.AI.Orchestrator` library from **62%** to **88%**.

## Specific Changes
Added unit tests for the following classes:
- OrchestratorRecognizer
- OrchestratorBotComponent

## Testing
The following image shows the new code coverage.
image

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.AI.Orchestrator.Tests/OrchestratorBotComponentTests.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Schema;
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.AI.Orchestrator.Tests/OrchestratorBotComponentTests.cs:4]: This line can be removed.

File: tests/Microsoft.Bot.Builder.AI.Orchestrator.Tests/OrchestratorRecognizerTest.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Moq;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Bot.Builder.AI.Orchestrator.Tests
+{
+    public class OrchestratorRecognizerTest
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.AI.Orchestrator.Tests/OrchestratorRecognizerTest.cs:16]: This class should be named OrchestratorRecognizerTest**s**

File: tests/Microsoft.Bot.Builder.AI.Orchestrator.Tests/OrchestratorRecognizerTest.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    {
+        [Fact]
+        public async Task LogsTelemetryThrowsArgumentNullExcetionOnNullDialogContext()
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.AI.Orchestrator.Tests/OrchestratorRecognizerTest.cs:19]:
```suggestion
        public async Task LogsTelemetryThrowsArgumentNullExceptionOnNullDialogContext()
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: tests/Microsoft.Bot.Builder.AI.Orchestrator.Tests/OrchestratorAdaptiveRecognizerTests.cs [hunk 4]
```diff
@@ -45,6 +46,80 @@ public async Task TestIntentRecognize()
...
+        }
+
+        [Fact]
+        public void TestOrchestratorRecognizerWithInvalidPath()
+        {
+            Assert.Throws<InvalidOperationException>(() => new OrchestratorRecognizer("C:/", "C:/")
+            {
+                ModelFolder = new StringExpression("C:/"),
+                SnapshotFile = new StringExpression("C:/")
+            });
```
Comment (sw-joelmut) [inline tests/Microsoft.Bot.Builder.AI.Orchestrator.Tests/OrchestratorAdaptiveRecognizerTests.cs:120]: I see you already passing the `ModelFolder` and `SnapshotFile` through the `OrchestratorRecognizer`, is the properties with the `new StringExpression("C:/")` instance even needed or vice-versa?

**Comment by sw-joelmut** [comment]
We should have a different naming for the branch when we promote it to MS. Additionally, for this case, we are creating a separate issue for each library we add tests and then relate that to the parent.
eg for naming and pr/issue: `https://github.com/microsoft/botbuilder-dotnet/pull/5890`
Also, the name of the PR has the `#390` at the end, we tend to put it at the start with [#XXX]
image

**Comment by MartinLuccanera** [comment]
Moved to a new PR https://github.com/southworks/botbuilder-dotnet/pull/394

**Pull request closed without merging**